src/app.rs

```diff
@@ -375,7 +370,26 @@ where
                     authenticated: Some(true),
                     actions: Some(lfs::Actions {
                         download: None,
-                        upload: Some(action),
+                        upload: Some(lfs::Action {
+                            href: storage
+                                .upload_url(
+                                    &StorageKey::new(
+                                        namespace.clone(),
+                                        object.oid,
+                                    ),
+                                    Duration::new(30 * 60, 0),
```

Comment:
Can this be extracted to a global constant so it isn't repeated in two places? For example:
```rust
/// The number of seconds the LFS client has before the upload URL expires.
const UPLOAD_EXPIRATION: Duration = Duration::from_secs(30 * 60);
```

`Duration::from_secs` is a `const fn`, so I think this should work.